indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java

```diff
@@ -78,96 +67,64 @@ public static ContactsAddressModel mapContactsAddress(CIAddressPropertyType2 add
-
-            addressItem.setCity(cityString.getCharacterString().getValue().toString());
+        var city = getNullIfNullPointer(() -> address.getCIAddress().getCity().getCharacterString().getValue().toString());
+        if (city != null) {
+            addressItem.setCity(city);
         }
-
-        CharacterStringPropertyType administrativeAreaString = address.getCIAddress().getAdministrativeArea();
-        if(administrativeAreaString != null
-                && administrativeAreaString.getCharacterString() != null
-                && administrativeAreaString.getCharacterString().getValue() != null) {
-            addressItem.setAdministrativeArea(administrativeAreaString.getCharacterString().getValue().toString());
+        var administrativeArea = getNullIfNullPointer(() -> address.getCIAddress().getAdministrativeArea().getCharacterString().getValue().toString());
+        if (administrativeArea != null) {
+            addressItem.setAdministrativeArea(administrativeArea);
         }
-
-        CharacterStringPropertyType postalCodeString = address.getCIAddress().getPostalCode();
-        if(postalCodeString != null
-                && postalCodeString.getCharacterString() != null
-                && postalCodeString.getCharacterString().getValue() != null) {
-            addressItem.setPostalCode(postalCodeString.getCharacterString().getValue().toString());
+        var postalCode = getNullIfNullPointer(() -> address.getCIAddress().getPostalCode().getCharacterString().getValue().toString());
+        if (postalCode != null) {
+            addressItem.setPostalCode(postalCode);
         }
-
-        CharacterStringPropertyType countryString = address.getCIAddress().getCountry();
-        if(countryString != null
-                && countryString.getCharacterString() != null
-                && countryString.getCharacterString().getValue() != null) {
-            addressItem.setCountry(countryString.getCharacterString().getValue().toString());
+        var country = getNullIfNullPointer(() -> address.getCIAddress().getCountry().getCharacterString().getValue().toString());
+        if (country != null) {
+            addressItem.setCountry(country);
         }
 
         return addressItem;
     }
 
     public static String mapContactsEmail(CharacterStringPropertyType electronicMailAddress) {
-
-        if(electronicMailAddress != null
-                && electronicMailAddress.getCharacterString() != null
-                && electronicMailAddress.getCharacterString().getValue() != null
-                && !"".equalsIgnoreCase(electronicMailAddress.getCharacterString().getValue().toString())) {
-            return electronicMailAddress.getCharacterString().getValue().toString();
-        }
-        else {
+        var email = getNullIfNullPointer(() -> electronicMailAddress.getCharacterString().getValue().toString());
+        if (email == null || email.trim().isEmpty()) {
             return null;
         }
+        return email;
```

Comment:
a bit strange, can you reverse the order of the if and return statements here, return email value if conditions are match, otherwise return null as default?

Reply:
I personally prefer inverting if statement, for better readable. If the `return null` precondition is a lot or complicated, putting return email at the end is better. For this situation, both ways are all good. So do you want me to reverse the order?

Reply:
for clearer, you should restrict the circumstance when a valid `email` value can be returned, the current logic will not guarantee that and lead to a false positive return, what if the `if` block you have got yet fully identified the cases (I understand for this case, those 2 conditions might be enough).

Reply:
general structure of returning something, not limited to just this `email` thing.

Reply:
after reversing: 

```java
        if (email != null && !email.trim().isEmpty()) {
            return email;
        }
        return null;
```

Actually it is doing the exactly same thing with the previous one. If previous one has a wrong return, the reversed version will also do.

Maybe this can better explain my point: https://www.jetbrains.com/help/resharper/InvertIf.html

Reply:
yep as I mentioned, the 2 conditions above for this `email` case might be enough therefore both orders will do the same. however, for general returning a value, for example

```
Type var = ...;
if (var == null || var is empty || <something else?>) return null;
return var;
```
will there be any cases that `var` is not null and also not empty but not a valid `var`,  if `<something else>` isnt defined in the `if`, it will accidentally return a non-valid `var`?

Reply:
Oh i think I've got your point. you mean may have something else need to check (such as whether a valid email with `@`, `.` etc )

If so, my code will need to add: 
```java
        if (email == null || email.trim().isEmpty()) {
            return null;
        }
        if (!isValid(email)) {
            return null;
        }
        return email;
```

reversed code also needs to update: 
```java
        if (email != null && !email.trim().isEmpty() && isValid(email)) {
            return email;
        }
        return null;
```

But anyway, I am also happy to use both ways. Just explained why i did it like this. 😃

Reply:
that is when you know exactly what needs to be added for `<something else>`, my concern is when you don't even realise there is a need for `<something else>`, as I keep saying, it is for general returning a value, not limited to `email`. You might know exactly the circumstances when a valid value should be returned, but you might don't fully aware of cases that lead to non-valid value.

Reply:
and if you miss to define such `<something else>`, the code will get a false positive return, the subsequent processes are allowed to run consuming invalid inputs.

Reply:
it is up to you at the end, this comment was part of the review decision that was ticked with `Comment` instead of `Request changes`.

Reply:
I think we are in the same way to thinking about the else conditions. Actually my way is also considering the circumstances when a valid value should be returned, and just do it in an inverted way. For example, a method needs to return a valid student. The student must have valid id, valid class, valid visa, and enough money. Common way is: 
```java
if (isValid(id) && isValid(class) && isValid(visa) && isValid(money)) {
    return student;
}
return null;
```
the code above means: **the student must comply all these conditions. Otherwise is invalid**.

On the other hand, the inverted way i was using is also considering: "the student must comply all these conditions, otherwise is invalid", and just in an inverted way to put these conditions.

So I think the ways we thinking are the same, just the ways we implementing are different.